From 6401d52cedb10455e4776d66e6b40f3fb197f538 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Wed, 29 Aug 2018 18:17:00 +0800
Subject: [PATCH 4502/5242] MLK-19380 driver: soc: update the noc QoS setting
 on imx8mq

commit  070173434d875093cc4d8e686da41ea20bd49125 from
https://source.codeaurora.org/external/imx/linux-imx.git

update the noc QoS setting for CPU & VPU on i.MX8MQ.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Reviewed-by: Jian Li <jian.li@nxp.com>
(cherry picked from commit 45d2dcaecce6d83e5c4a7e9488c651a05b0f05ac)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/soc-imx8.c |   26 ++++++++------------------
 include/soc/imx/fsl_sip.h  |    6 ++++++
 2 files changed, 14 insertions(+), 18 deletions(-)

diff --git a/drivers/soc/imx/soc-imx8.c b/drivers/soc/imx/soc-imx8.c
index 2b28269..0fc298e 100644
--- a/drivers/soc/imx/soc-imx8.c
+++ b/drivers/soc/imx/soc-imx8.c
@@ -280,29 +280,19 @@ static ssize_t imx8_get_soc_uid(struct device *dev,
 
 static void __init imx8mq_noc_init(void)
 {
-	struct device_node *np;
-	const char *status;
-	int statlen;
 	struct arm_smccc_res res;
 
-	np = of_find_compatible_node(NULL, NULL, "fsl,imx8mq-lcdif");
-	if (!np)
-		return;
-
-	status = of_get_property(np, "status", &statlen);
-	if (status == NULL)
-		return;
+	pr_info("Config NOC for VPU and CPU\n");
 
-	if (statlen > 0) {
-		if (!strcmp(status, "disabled"))
-			return;
-	}
+	arm_smccc_smc(FSL_SIP_NOC, FSL_SIP_NOC_PRIORITY, NOC_CPU_PRIORITY,
+			0x80000300, 0, 0, 0, 0, &res);
+	if (res.a0)
+		pr_err("Config NOC for CPU fail!\n");
 
-	pr_info("Config NOC for VPU and CPU\n");
-	arm_smccc_smc(FSL_SIP_NOC, FSL_SIP_NOC_LCDIF, 0,
-			0, 0, 0, 0, 0, &res);
+	arm_smccc_smc(FSL_SIP_NOC, FSL_SIP_NOC_PRIORITY, NOC_VPU_PRIORITY,
+			0x80000300, 0, 0, 0, 0, &res);
 	if (res.a0)
-		pr_err("Config NOC for VPU and CPU fail!\n");
+		pr_err("Config NOC for VPU fail!\n");
 }
 
 static int __init imx8_soc_init(void)
diff --git a/include/soc/imx/fsl_sip.h b/include/soc/imx/fsl_sip.h
index f1c8c9b..44649f4 100644
--- a/include/soc/imx/fsl_sip.h
+++ b/include/soc/imx/fsl_sip.h
@@ -59,6 +59,12 @@
 
 #define FSL_SIP_NOC			0xc2000008
 #define FSL_SIP_NOC_LCDIF		0x0
+#define FSL_SIP_NOC_PRIORITY		0x1
+#define NOC_GPU_PRIORITY		0x10
+#define NOC_DCSS_PRIORITY		0x11
+#define NOC_VPU_PRIORITY		0x12
+#define NOC_CPU_PRIORITY		0x13
+#define NOC_MIX_PRIORITY		0x14
 
 #define FSL_SIP_WAKEUP_SRC		0xc2000009
 #define FSL_SIP_WAKEUP_SRC_SCU		0x1
-- 
1.7.9.5

